nucliadb/nucliadb/common/maindb/utils.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+from nucliadb.common.maindb.driver import Driver
+from nucliadb.ingest.settings import settings
+from nucliadb_utils.exceptions import ConfigurationError
+from nucliadb_utils.store import MAIN
+
+try:
+    from nucliadb.common.maindb.redis import RedisDriver
+
+    REDIS = True
+except ImportError:  # pragma: no cover
+    REDIS = False
+
+try:
+    from nucliadb.common.maindb.tikv import TiKVDriver
+
+    TIKV = True
+except ImportError:  # pragma: no cover
+    TIKV = False
+
+
+try:
+    from nucliadb.common.maindb.pg import PGDriver
+
+    PG = True
+except ImportError:  # pragma: no cover
+    PG = False
+
+try:
+    from nucliadb.common.maindb.local import LocalDriver
+
+    FILES = True
+except ImportError:  # pragma: no cover
+    FILES = False
+
+_DRIVER_UTIL_NAME = "driver"
+
+
+async def get_driver() -> Driver:
```

Comment:
(doubt) Why are we lazily creating maindb driver? Other utilities are created and set at startup time. Lazy creation adds dependency with settings.

Reply:
I agree and think it's worth while to fix this as well.